src/App.tsx

```diff
@@ -8,53 +8,116 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { getPosts, getUsers } from './api/api';
+import { Post } from './types/Post';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [activeUser, setActiveUser] = useState<User | null>(null);
+  const [activePost, setActivePost] = useState<Post | null>(null);
+  const [showSideBar, setShowSideBar] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(() => setIsError(true));
```

Comment:
Consider providing more detailed error handling or logging in the catch block to help with debugging and understanding what went wrong.